src/remoted/request.c

```diff
@@ -161,11 +169,15 @@ void * req_dispatch(req_node_t * node) {
 
     for (attempts = 0; attempts < max_attempts; attempts++) {
         // Try to send message
-        if (send_msg(agentid, payload, ldata) < 0) {
+        ldata = isForceReconnect ? strlen(msg_to_send) : ldata;
+        if (send_msg(agentid, (isForceReconnect == true) ? msg_to_send : payload, ldata) < 0) {
             merror("Cannot send request to agent '%s'", agentid);
             OS_SendSecureTCP(node->sock, strlen(WR_SEND_ERROR), WR_SEND_ERROR);
             goto cleanup;
         } else {
+            if (OS_SendSecureTCP(node->sock, strlen("ack"), "ack") < 0) {
```

Comment:
Instead this code, I would add this code [here](https://github.com/wazuh/wazuh/pull/15281/files#diff-d03df09093a8316731cbee56730a13bed4c9b7c8f1d14d28481931f8beb1b671R183) so it only applies only to `force_reconnect`:
```c
if (isForceReconnect) {
    if (OS_SendSecureTCP(node->sock, strlen("ack"), "ack") != 0) {
        mwarn("At OS_SendSecureTCP(): %s", strerror(errno));
    }
    goto cleanup;
}
```